bundles/org.openhab.binding.powermax/src/main/java/org/openhab/binding/powermax/internal/message/PowermaxPowerlinkMessage.java

```diff
@@ -50,37 +48,29 @@ public PowermaxState handleMessage(PowermaxCommManager commManager) {
 
         if (subType == 0x03) {
             // keep alive message
+
+            addDebugInfo("Subtype", subType, "Keep Alive");
+
             commManager.sendAck(this, (byte) 0x02);
             updatedState = commManager.createNewState();
             updatedState.setLastKeepAlive(System.currentTimeMillis());
-        } else if (subType == 0x0A && message[4] == 0x01) {
-            logger.debug("Powermax alarm binding: Enrolling Powerlink");
-            commManager.enrollPowerlink();
-            updatedState = commManager.createNewState();
-            updatedState.setDownloadSetupRequired(true);
-        } else {
-            commManager.sendAck(this, (byte) 0x02);
-        }
-
-        return updatedState;
-    }
-
-    @Override
-    public String toString() {
-        String str = super.toString();
+        } else if (subType == 0x0A) {
+            byte enroll = message[4];
 
-        byte[] message = getRawData();
-        byte subType = message[2];
+            addDebugInfo("Subtype", subType, "Enroll");
+            addDebugInfo("Enroll", enroll);
 
-        if (subType == 0x03) {
-            str += "\n - sub type = keep alive";
-        } else if (subType == 0x0A) {
-            str += "\n - sub type = enroll";
-            str += "\n - enroll = " + String.format("%02X", message[4]);
+            if (enroll == 0x01) {
+                logger.debug("Powermax alarm binding: Enrolling Powerlink");
+                commManager.enrollPowerlink();
```

Comment:
Here is the case while ACK is not sent.
It is where your separation between handleMessage and handleMessageInternal was not a good idea.